lib/internal/streams/readable.js

```diff
@@ -101,7 +101,7 @@ function ReadableState(options, stream, isDuplex) {
   // Note: 0 is a valid value, means "don't call _read preemptively ever"
   this.highWaterMark = options ?
     getHighWaterMark(this, options, 'readableHighWaterMark', isDuplex) :
-    getDefaultHighWaterMark(false);
```

Comment:
This was added in Hmm it was added in https://github.com/nodejs/node/commit/bd02775af8db478c3564d4ca44ec171e403547df and I think it makes sense. If `options` is falsy it means the stream is not in object mode, so `false` is used.